From 272a6c83642cdd3825c442c2e612f037ebfe366f Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Thu, 17 May 2018 23:18:00 +0000
Subject: [PATCH] Fix incorrect handling of symlinks in nc_rm_rf

Update nftw flags to pass FTW_PHYS so that valid symlinks will not
resolve to their target files. Without this flag, the _rm_rf callback
can get a typeflag for a folder and wrongly cause an attempt to run
rmdir on a symlink.
---
 src/files.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/files.c b/src/files.c
index 4f1c505..40a1b4f 100644
--- a/src/files.c
+++ b/src/files.c
@@ -78,7 +78,7 @@ static int _rm_rf(const char *path, __attribute__((unused)) const struct stat *s
 
 bool nc_rm_rf(const char *path)
 {
-        bool ret = nftw(path, _rm_rf, 35, FTW_DEPTH) == 0;
+        bool ret = nftw(path, _rm_rf, 35, FTW_DEPTH | FTW_PHYS) == 0;
 
         sync();
         return ret;
